src/main/java/graphql/relay/Relay.java

```diff
@@ -172,23 +172,39 @@ public GraphQLObjectType connectionType(String name, GraphQLObjectType edgeType,
                 .build();
     }
 
-
     public GraphQLFieldDefinition mutationWithClientMutationId(String name, String fieldName,
                                                                List<GraphQLInputObjectField> inputFields,
                                                                List<GraphQLFieldDefinition> outputFields,
                                                                DataFetcher dataFetcher) {
+        GraphQLInputObjectField clientMutationIdInputField = newInputObjectField()
+                .name("clientMutationId")
+                .type(GraphQLString)
+                .build();
+        GraphQLFieldDefinition clientMutationIdPayloadField = newFieldDefinition()
+                .name("clientMutationId")
+                .type(GraphQLString)
+                .build();
+
+        return mutation(name, fieldName, addElementToList(inputFields, clientMutationIdInputField), 
+                addElementToList(outputFields, clientMutationIdPayloadField), dataFetcher);
+    }
+    
+    private static <T> List<T> addElementToList(List<T> list, T element) {
+        ArrayList<T> result = new ArrayList<>(list);
+        result.add(element);
+        return result;
+    }
```

Comment:
This can be `Collections.singletonList` or `Arrays.asList()`

Reply:
I'm not sure I understand?

This method exists to take a `List` of (input/payload) fields and add the `clientMutationId` field. But the given `List` may be immutable, so we need to copy its elements to a new `List` that lets us add the `clientMutationId` field.